distribution/lib/Standard/Base/0.0.0-dev/src/Internal/Ordering_Helpers.enso

```diff
@@ -69,16 +69,16 @@ assert_same_comparators this that ~action =
         False -> Error.throw (Incomparable_Values.Error this that)
 
 compare_with_comparators this that -> Ordering ! Incomparable_Values =
-    any_result = Panic.catch Type_Error handler=identity <|
+    yield_an_error _ = Error.throw (Incomparable_Values.Error this that)
+    Panic.catch Type_Error handler=yield_an_error <|
         comp_this = Comparable.from this
         comp_that = Comparable.from that
 
-        if Meta.is_same_object comp_this.comparator comp_that.comparator then
+        any_result = if Meta.is_same_object comp_this.comparator comp_that.comparator then
             comp_this.comparator.compare comp_this.value comp_that.value
 
-    case any_result of
-        result:Ordering -> result
-        _ -> Error.throw (Incomparable_Values.Error this that)
+        result = any_result:Ordering
+        result
```

Comment:
Do I understand correctly that the control flow in the negative case here is:
1. the `if` has no `else` so it will return `Nothing` and assign it to `any_result`
2. the `any_result:Ordering` ascription will then fail with `Type_Error`
3. that is caught in the `Panic.catch` and turned into the `Incomparable_Values` dataflow error
?

I'm not a fan of control flow jumping around like that. If that's what it takes to make this fast - OK.

But what if we do:
```
if Meta.is_same_object comp_this.comparator comp_that.comparator then (comp_this.comparator.compare comp_this.value comp_that.value):Ordering else Error.throw (Incomparable_Values.Error this that)
```
?
(that would look much better with #6408)

Reply:
It's not just about how the code looks (and alas your version has too long line), but also how fast it can run. Moreover I wanted to avoid duplicated `Error.throw`. I could write the code as
```ruby
compare_with_comparators this that -> Ordering ! Incomparable_Values =
    yield_an_error _ = Error.throw (Incomparable_Values.Error this that)
    Panic.catch Type_Error handler=yield_an_error <|
        comp_this = Comparable.from this
        comp_that = Comparable.from that

        result = (if Meta.is_same_object comp_this.comparator comp_that.comparator then comp_this.comparator.compare comp_this.value comp_that.value else yield_an_error Nothing):Ordering
        result
```
but such a version is slower, so I won't.

Reply:
Ok